Source/WebCore/PAL/pal/spi/cocoa/CoreCryptoSPI.h

```diff
@@ -0,0 +1,101 @@
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+WTF_EXTERN_C_BEGIN
+
+#if USE(APPLE_INTERNAL_SDK)
+#include <corecrypto/ccec25519.h>
+#include <corecrypto/ccec25519_priv.h>
+#include <corecrypto/ccsha2.h>
+#else
+
+#define CC_SPTR(_sn_, _n_) _n_
+
+#define CC_WIDE_NULL nullptr
+
+#define CCRNG_STATE_COMMON \
+int(*CC_SPTR(ccrng_state, generate))(struct ccrng_state *rng, size_t outlen, void *out);
+
+struct ccrng_state {
+    CCRNG_STATE_COMMON
+};
+struct ccrng_state *ccrng(int *error);
+
+#define ccrng_generate(rng, outlen, out) \
+((rng)->generate((struct ccrng_state *)(rng), (outlen), (out)))
+
+typedef uint8_t ccec25519key[32];
+typedef ccec25519key ccec25519secretkey;
+typedef ccec25519key ccec25519pubkey;
+typedef ccec25519key ccec25519base;
+
+typedef uint8_t ccec25519signature[64];
+
+const struct ccdigest_info *ccsha512_di(void);
+#define ccoid_sha512 ((unsigned char *)"\x06\x09\x60\x86\x48\x01\x65\x03\x04\x02\x03")
+#define ccoid_sha512_len 11
+/* SHA512 */
+#define CCSHA512_BLOCK_SIZE  128
+#define    CCSHA512_OUTPUT_SIZE  64
+#define    CCSHA512_STATE_SIZE   64
+extern const struct ccdigest_info ccsha512_ltc_di;
+
+void cccurve25519(ccec25519key out, const ccec25519secretkey sk, const ccec25519base);
+inline void cccurve25519_make_priv(struct ccrng_state *rng, ccec25519secretkey sk)
+{
+    ccrng_generate(rng, 32, sk);
+    sk[0] &= 248;
+    sk[31] &= 127;
+    sk[31] |= 64;
+}
+
+inline void cccurve25519_make_pub(ccec25519pubkey pk, const ccec25519secretkey sk)
+{
+    cccurve25519(pk, sk, CC_WIDE_NULL);
+}
+
+inline void cccurve25519_make_key_pair(struct ccrng_state *rng, ccec25519pubkey pk, ccec25519secretkey sk)
+{
+    cccurve25519_make_priv(rng, sk);
+    cccurve25519_make_pub(pk, sk);
+}
+
+int cced25519_make_pub(const struct ccdigest_info *, ccec25519pubkey pk, const ccec25519secretkey sk);
+
+void cced25519_make_key_pair(const struct ccdigest_info *, struct ccrng_state *, ccec25519pubkey pk, ccec25519secretkey sk);
+#if __has_feature(bounds_attributes)
+#define cc_sized_by(x) __attribute__((sized_by(x)))
+#else
+#define cc_sized_by(x)
+#endif // __has_feature(bounds_attributes)
+
+void cced25519_sign(const struct ccdigest_info *, ccec25519signature, size_t len, const void *cc_sized_by(len) msg, const ccec25519pubkey pk, const ccec25519secretkey sk);
+int cced25519_verify(const struct ccdigest_info *, size_t len, const void *cc_sized_by(len) msg, const ccec25519signature, const ccec25519pubkey pk);
```

Comment:
This seems very similar to a file you are adding in https://github.com/WebKit/WebKit/pull/5026.
Let's focus on one PR. Once it is finalised, we can work on the other.